Modules/hpe3par_volume.py

```diff
@@ -642,6 +642,11 @@ def convert_type(
             {})
     try:
         client_obj.login(storage_system_username, storage_system_password)
+        compression_state = client_obj.getVolume(volume_name).compression_state
+        if compression_state == 2 or compression_state == 3:
+            compression_state = False
+        else:
+            compression_state = True
```

Comment:
Volume compressionState enumeration
Symbol Value Description
YES 1 Compression is enabled on the volume.
NO 2 Compression is disabled on the volume.
OFF 3 Compression is turned off.
NA 4 Compression is not available on the volume.

@ansarars Did you check any volume which has compression_state set to 4? Should we set the state to True in such case?

Reply:
I have added compression_state 4 for volume type having provisioning type full. I have tested it by changing volume type from thin to full, full to full and full to full with compression